idseq_pipeline/commands/common.py

```diff
@@ -27,13 +27,72 @@
 MAX_CONCURRENT_COPY_OPERATIONS = 8
 iostream = threading.Semaphore(MAX_CONCURRENT_COPY_OPERATIONS)
 
+
+class StatsFile(object):
+
+    def __init__(self, stats_filename, local_results_dir, s3_input_dir, s3_output_dir):
+        self.local_results_dir = local_results_dir
+        self.stats_filename = stats_filename
+        self.s3_input_dir = s3_input_dir
+        self.s3_output_dir = s3_output_dir
+        self.stats_path = os.path.join(self.local_results_dir, self.stats_filename)
+        self.data = []
+        self.mutex = threading.RLock()
+
+    def load_from_s3(self):
+        stats_s3_path = os.path.join(self.s3_input_dir, self.stats_filename)
+        if check_s3_file_presence(stats_s3_path):
+            execute_command("aws s3 cp --quiet %s %s/" % (stats_s3_path, self.local_results_dir))
+        self._load()
+
+    def _load(self):
+        if os.path.isfile(self.stats_path):
+            with open(self.stats_path) as f:
+                self.data = json.load(f)
+
+    def _save(self):
+        with open(self.stats_path, 'wb') as f:
+            json.dump(self.data, f)
+
+    def save_to_s3(self):
+        with self.mutex:
+            self._save()
+            execute_command("aws s3 cp --quiet %s %s/" % (self.stats_path, self.s3_output_dir))
+
+    def get_total_reads(self):
```

Comment:
Note:  I didn't write most of the code in this class, just pulled it in from already existing functions.